src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,79 @@
-export const Pagination = () => {};
+import React from 'react';
+import cn from 'classnames';
+
+type Props = {
+  total: number;
+  itemsPerPage: number;
+  currentPage: number;
+  onPageChange: (currentPage: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  itemsPerPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberOfPages = Math.ceil(total / itemsPerPage);
+  const pages = Array(numberOfPages).fill(0).map((_, i) => i + 1);
+  const isPrevAreaDisabled = currentPage === 1;
+  const isNextAreaDisabled = currentPage === numberOfPages;
+
+  const handleClickToNext = (nextPage: number) => {
+    return (currentPage !== numberOfPages
+      && (onPageChange(nextPage)));
+  };
+
+  const handleClickToPrev = (previousPage: number) => {
+    return (currentPage !== 1
+      && (onPageChange(previousPage)));
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={cn('page-item', {
+        disabled: isPrevAreaDisabled,
+      })}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={isPrevAreaDisabled}
+          onClick={() => handleClickToPrev(currentPage - 1)}
```

Comment:
If you divide the prev and next click into two separate functions, then you can do calculations in the middle of each handle and not pass anything to the arguments.